ec2/spark_ec2.py

```diff
@@ -872,9 +890,16 @@ def deploy_files(conn, root_dir, opts, master_nodes, slave_nodes, modules):
     if "." in opts.spark_version:
         # Pre-built Spark deploy
         spark_v = get_validate_spark_version(opts.spark_version, opts.spark_git_repo)
+        tachyon_v = get_tachyon_version(spark_v)
     else:
         # Spark-only custom deploy
         spark_v = "%s|%s" % (opts.spark_git_repo, opts.spark_version)
+        if opts.tachyon_version is None:
```

Comment:
Yeah this seems like an unnecessary restriction for users who just want to change the Spark git hash. 

Given that users who use the git hash are going to be advanced users why don't we just skip version checks for it ? If users want to use a specific git hash I think its fine to expect them to figure out which tachyon version is required or not etc.